src/Gui/DlgUnitsCalculatorImp.cpp

```diff
@@ -201,11 +202,11 @@ void DlgUnitsCalculator::onUnitsBoxActivated(int index)
     // SI units use [m], not [mm] for lengths
     //
     Base::Quantity q = ui->quantitySpinBox->value();
-    int32_t old = q.getUnit().getSignature().Length;
+    int32_t old = q.getUnit()[0];
```

Comment:
Is similar to boolean blindness because the meaning of 0 is not obvious. As said above there should be a method length() to make the meaning clear.

Reply:
...such a methods implemented now.